data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/GrokLogstashPluginAttributesMapper.java

```diff
@@ -6,100 +6,52 @@
 package org.opensearch.dataprepper.logstash.mapping;
 
 import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
-import org.opensearch.dataprepper.logstash.model.LogstashValueType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-import java.util.stream.Collectors;
 import java.util.Arrays;
-import java.util.List;
-import java.util.ArrayList;
-import java.util.Map;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 
 class GrokLogstashPluginAttributesMapper extends AbstractLogstashPluginAttributesMapper {
+    private final GrokLogstashPluginAttributeMapperHelper matchAttributesHelper;
+    private final GrokLogstashPluginAttributeMapperHelper patternDefinitionsAttributeHelper;
+    private final GrokLogstashPluginAttributeMapperHelper overwriteAttributeHelper;
+
     protected static final String LOGSTASH_GROK_MATCH_ATTRIBUTE_NAME = "match";
     protected static final String LOGSTASH_GROK_PATTERN_DEFINITIONS_ATTRIBUTE_NAME = "pattern_definitions";
-    private static final Logger LOG = LoggerFactory.getLogger(GrokLogstashPluginAttributesMapper.class);
+    protected static final String LOGSTASH_GROK_OVERWRITE_ATTRIBUTE_NAME = "overwrite";
+    private static  final HashSet<String> customAttributes = new HashSet<>(Arrays.asList(LOGSTASH_GROK_MATCH_ATTRIBUTE_NAME,
+            LOGSTASH_GROK_PATTERN_DEFINITIONS_ATTRIBUTE_NAME,
+            LOGSTASH_GROK_OVERWRITE_ATTRIBUTE_NAME));
+
+    public GrokLogstashPluginAttributesMapper() {
+        matchAttributesHelper = new GrokMatchAttributeHelper();
+        patternDefinitionsAttributeHelper = new GrokPatternDefinitionsAttributeHelper();
+        overwriteAttributeHelper = new GrokOverwriteAttributeHelper();
+    }
 
-    @SuppressWarnings("unchecked")
     @Override
     protected void mapCustomAttributes(List<LogstashAttribute> logstashAttributes, LogstashAttributesMappings logstashAttributesMappings, Map<String, Object> pluginSettings) {
-        final List<LogstashAttribute> matchAttributes = new ArrayList<>();
-        final Map<String, String> patternDefinitions = new HashMap<>();
+        final Map<String, GrokLogstashPluginAttributeMapperHelper> helperInstances = new HashMap<>();
+        helperInstances.put(LOGSTASH_GROK_MATCH_ATTRIBUTE_NAME, matchAttributesHelper);
+        helperInstances.put(LOGSTASH_GROK_PATTERN_DEFINITIONS_ATTRIBUTE_NAME, patternDefinitionsAttributeHelper);
+        helperInstances.put(LOGSTASH_GROK_OVERWRITE_ATTRIBUTE_NAME, overwriteAttributeHelper);
```

Comment:
This can be moved to constructor as well. 

Define helperInstances as class instance variable and initialize in the constructor.
```private final Map<String, GrokLogstashPluginAttributeMapperHelper> helperInstances```